warehouse/packaging/models.py

```diff
@@ -265,6 +268,15 @@ def __table_args__(cls):  # noqa
     )
     classifiers = association_proxy("_classifiers", "classifier")
 
+    @property
+    def structured_classifiers(self):
+        structured = defaultdict(list)
+        for classifier in self.classifiers:
+            key, *value = classifier.split(' :: ', 1)
+            if value:
+                structured[key].append(value[0])
+        return structured
```

Comment:
This is pretty similar to what's happening here (though not exactly the same):
https://github.com/pypa/warehouse/blob/32b27e77c5caaf2a0aafe450f200a076b3135c24/warehouse/views.py#L210-L214

I wonder if there might be a way to share this functionality between both places.